Assets/Scripts/DynamicCamera/CameraBehavior.cs

```diff
@@ -0,0 +1,76 @@
+﻿namespace HarmonyQuest.DynamicCamera
+{
+    using UnityEngine;
+
+    public abstract class CameraBehavior : MonoBehaviour
+    {
+        [SerializeField]
+        private TestPlayer player;
+        private CharacterController characterController;
+        
+        protected Vector3 targetAngles;
+
+        protected float bias = 1f;
+        protected bool active = false;
+        [SerializeField]
+        protected Vector3 direction;
+
+        protected void Init()
+        {
+            characterController = player.GetComponent<CharacterController>();
+        }
+
+        public void Move()
+        {
+            if (!active)
+                return;
+            if (!Vector3.Equals(direction, Vector3.zero))
+            {
+                transform.position = Vector3.Lerp(transform.position, direction, bias * Time.deltaTime);
+            }
+            transform.eulerAngles = Vector3.Lerp(transform.eulerAngles, targetAngles, bias * Time.deltaTime);
+        }
+
+        public void ToggleActive()
```

Comment:
I think it'd be better to have this function accept a bool. That way we know 100% what state the camera will enter.

Reply:
You got it chief